lib/ansible/modules/cloud/memset/memset_memstore_user.py

```diff
@@ -0,0 +1,334 @@
+            retvals['changed'] = True
+            retvals['memset_api'] = response.json()
+    else:
+        '''
+        User exists, update it. The only user attribute we can change is to enable or disable.
+        The password will be changed if update_password is set to `always`.
+        '''
+        if args['update_password'] == 'always':
+            _set_password = True
+
+        if user['enabled'] == args['enabled']:
+            if args['check_mode']:
+                return(retvals)
+        else:
+            if args['check_mode']:
+                retvals['changed'] = True
+                return(retvals)
+
+            payload['name'], payload['username'] = args['memstore'], args['username']
+
+            if args['enabled']:
+                api_method = 'memstore.user.enable'
+            else:
+                api_method = 'memstore.user.disable'
+
+            retvals['failed'], msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+
+            if retvals['failed']:
+                retvals['msg'] = msg
+            else:
+                retvals['changed'] = True
+                payload['enabled'] = args['enabled']
+                retvals['memset_api'] = payload
+
+    '''
+    If update_password is set to `always` then the task will always submit the current password to the
+    API which does not expose if it has been changed - setting the same password results in a successful
+    operation. Because of this, tasks will always return changed even if the password remains the same.
+    '''
+    if _set_password and not retvals['failed']:
+        pwd_retvals = update_password(args=args)
+        retvals['password'] = pwd_retvals['password']
+        if pwd_retvals['changed']:
+            retvals['changed'] = True
+
+    return(retvals)
+
+
+def delete_user(args=None, user=None):
+    '''
+    Deletes a user.
+    '''
+    retvals, payload = dict(), dict()
+    retvals['changed'], retvals['failed'] = False, False
+
+    if user:
+        if args['check_mode']:
+            retvals['changed'] = True
+            return(retvals)
+        payload['name'] = args['memstore']
+        payload['username'] = args['username']
+        api_method = 'memstore.user.delete'
+        retvals['failed'], msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+
+        if retvals['failed']:
+            retvals['msg'] = msg
+        else:
+            retvals['changed'] = True
+
+    return(retvals)
+
+
+def update_password(args=None):
+    '''
+    Updates the password of an existing user.
+    '''
+    retvals, payload = dict(), dict()
+    retvals['changed'], retvals['failed'] = False, False
+
+    if args['check_mode']:
+        # Updating a password will always return changed as the API does not
+        # know if the password is actually being changed.
+        retvals['password'] = 'Check mode cannot tell if the password will be changed.'
```

Comment:
hmm using the key 'password' is not a good idea for a msg to the user, ansible would think this might be sensible data.  (the return password would also be missing in the return docs) however, I would suggest to use the common return key `msg`